docs/toolhive/guides-registry/deployment.mdx

```diff
@@ -0,0 +1,308 @@
+---
+title: Deployment
```

Comment:
@ChrisJBurns - hey, can you help review that part?

@blkt - should we mention the authentication options too (how to disable/enable/configure it)